### ISSUE TITLE
[Accessibility]: Previous page meta title is read when link is clicked

### ISSUE THREAD

**Issue by st0nebraker** [sourcegraph/about#5534]
### Audit type

Screen reader navigation

### User journey audit issue

#5500

### Problem description

While using a screen reader, if I click a link, the previous page meta title is read, instead of the current/new page. This is confusing to a user.

Steps to reproduce:
1. Navigate to home (about.sourcegraph.com)
2. Turn on your [screen reader](https://docs.sourcegraph.com/dev/background-information/web/accessibility/how-to-screen-reader).
3. Click into `Request a demo` button link.
4. Notice "Sourcegraph | Code Intelligence Platform" is read first, instead of "Request a Demo | Sourcegraph".

1. Navigate to (about.sourcegraph.com/code-insights)
2. Turn on your [screen reader](https://docs.sourcegraph.com/dev/background-information/web/accessibility/how-to-screen-reader).
3. Click into `Request a demo` button link.
4. Notice "Sourcegraph | Code Insights" is read first, instead of "Request a Demo | Sourcegraph".

### Expected behavior

As a user utilizing a screen reader, I expect button actions to focus on the correct next item so I understand my action was completed. When a link is clicked, the current/new page meta title is what should be picked up by screen readers.

### Assigning labels

- [ ] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [ ] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
